modules/javafx.controls/src/main/java/javafx/scene/control/skin/VirtualFlow.java

```diff
@@ -1894,8 +1917,11 @@ T getLastVisibleCellWithinViewPort() {
         return null;
     }
 
-    // Returns first visible cell whose bounds are entirely within the viewport
-    T getFirstVisibleCellWithinViewPort() {
+    /**
+     *
+     * @return first visible cell whose bounds are entirely within the viewport
+     */
+    protected T getFirstVisibleCellWithinViewPort() {
```

Comment:
These two methods getFirstVisibleCellWithinViewPort() and getLastVisibleCellWithinViewPort() are very useful when someone manually insert some rows into the VirtualFlow. One may needs to know which index are actually visible to the end user in the viewport.